src/session.ts

```diff
@@ -22,36 +30,117 @@ export type State =
   | { connection: Connection.Connecting }
   | { connection: Connection.Connected; session: Session };
 
+export type ConnectionModalState =
+| { status: ConnectionModalStateType.Open; modalProps: ConnectionModalProps }
+| { status: ConnectionModalStateType.Close; modalProps: null };
 export interface Session {
   address: string;
   tokenBalance: BigNumber | null; // `null` means it isn't loaded yet.
   tx: TxState;
 }
-
+export interface ConnectionModalProps {
+  config: Config;
+  uri: string;
+}
 export interface Store extends Readable<State> {
-  connect(config: Config): Promise<void>;
+  connectMetamask(config: Config): Promise<void>;
   updateBalance(n: BigNumber): void;
   refreshBalance(config: Config): Promise<void>;
-
+  provider: ethers.providers.Provider;
+  signer: ethers.Signer;
+  connectWalletConnect(config: Config): Promise<void>;
   setTxSigning(): void;
   setTxPending(tx: TransactionResponse): void;
   setTxConfirmed(tx: TransactionReceipt): void;
   setChangedAccount(address: string): void;
 }
+let walletConnect: any;
```

Comment:
Why is there another walletConnect global here when you have in Config? Why do you need two?